recipes/aws-libfabric/all/conanfile.py

```diff
@@ -0,0 +1,113 @@
+    homepage = "https://github.com/aws/libfabric"
+    license = "BSD-2-Clause", "GPL-2.0-or-later"
+    settings = "os", "arch", "compiler", "build_type"
+    _providers = ['gni', 'psm', 'psm2', 'sockets', 'rxm', 'tcp', 'udp', 'usnic', 'verbs', 'bgq', 'shm', 'efa', 'rxd', 'mrail', 'rstream', 'perf', 'hook_debug']
+    options = {
+        **{ p: "ANY" for p in _providers },
+        **{
+            "shared": [True, False],
+            "fPIC": [True, False],
+            "with_libnl": [True, False],
+            "with_bgq_progress": [None, "auto", "manual"],
+            "with_bgq_mr": [None, "basic", "scalable"]
+        }
+    }
+    default_options = {
+        **{ p: "auto" for p in _providers if p not in ('efa', 'verbs', 'rxd') },
+        **{
+            "shared": False,
+            "fPIC": True,
+            "efa": "yes",
+            "verbs": "no",
+            "rxd": "no",
+            "with_libnl": False,
+            "with_bgq_progress": None,
+            "with_bgq_mr": None
+        }
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    _autotools = None
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def requirements(self):
+        if self.options.get_safe('with_libnl'):
+            self.requires('libnl/3.2.25')
+
+    def validate(self):
+        if self.settings.compiler == "Visual Studio":
+            raise ConanInvalidConfiguration("The libfabric package cannot be built on Visual Studio.")
+        for p in self._providers:
+            if self.options.get_safe(p) not in ["auto", "yes", "no", "dl"] and not os.path.isdir(str(self.options.get_safe(p))):
+                raise ConanInvalidConfiguration("Option {} can only be one of 'auto', 'yes', 'no', 'dl' or a directory path")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self)
+        with tools.chdir(self._source_subfolder):
+            self.run("./autogen.sh", win_bash=tools.os_info.is_windows)
+        args = []
+        for p in self._providers:
+            args.append('--enable-{}={}'.format(p, self.options.get_safe(p)))
+        if self.options.with_libnl:
+            args.append('--with-libnl={}'.format(self.deps_cpp_info["libnl"].rootpath))
+        if self.options.with_bgq_progress:
+            args.append('--with-bgq-progress={}'.format(self.options.with_bgq_progress))
+        if self.options.with_bgq_mr:
+            args.append('--with-bgq-mr={}'.format(self.options.with_bgq_mr))
```

Comment:
This assumes `configure` accepts `no`.
```suggestion
        yes_no_dl = lambda v: {"True": "yes", "False": "no", "dl": "dl"}[str(v)]
        value_no = lambda v: {"None": "no", "False": "no"}.get(str(v)]
        args = [
            "--with-bgq-progress={}".format(value_no(self.options.with_bgq_progress)),
            "--with-bgq-mr={}".format(value_no(self.options.with_bgq_mr)),
        ]
        for p in self._providers:
            args.append("--enable-{}={}".format(p, yes_no_dl(getattr(self.options, p))))
        if self.options.with_libnl:
            args.append("--with-libnl={}".format(tools.unix_path(self.deps_cpp_info["libnl"].rootpath))),
        else:
            args.append("--with-libnl=no")
```